### PR TITLE
Expose applications that need support intervention

### PR DESCRIPTION
**Pull request by tijmenb**
## Context

There are ~3~ 4 situations in which applications need manual intervention:

- The course is removed from Apply
- The course is removed from Find
- The course's site is removed from Find
- The course is full

## Changes proposed in this pull request

Add a page with applications that need intervention. It's fairly MVP! Also a few UI tweaks to make it easier to navigate the support section.

Screenshot 2020-04-06 at 14 44 15

## Guidance to review

Does it make sense?

## Link to Trello card

https://trello.com/c/fRAKJt36/1264-expose-applications-that-need-support-intervention

## Things to check

- [x] This code doesn't rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/services/support_interface/application_monitor.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+module SupportInterface
+  class ApplicationMonitor
```
Comment (duncanjbrown) [inline app/services/support_interface/application_monitor.rb:2]: Once an application has finished its life — ie enrolled, declined, withdrawn, etc, should we filter it out of these queries?

File: app/models/course.rb [hunk 2]
```diff
@@ -55,4 +55,8 @@ def both_study_modes_available?
   def full?
     course_options.all?(&:no_vacancies?)
   end
+
+  def find_url
```
Comment (duncanjbrown) [inline app/models/course.rb:59]: 🎉

Comment (tvararu) [reply app/models/course.rb:59]: Long overdue 👍

File: app/services/support_interface/application_monitor.rb (hunk 1 above)
Comment (tijmenb) [reply app/services/support_interface/application_monitor.rb:2]: Yeah, that's a good point. Pushed a commit.

File: app/models/course.rb [hunk 3]
```diff
@@ -55,4 +55,8 @@ def both_study_modes_available?
...
+    @find_url ||= "https://www.find-postgraduate-teacher-training.service.gov.uk/course/#{provider.code}/#{code}"
```
Comment (tvararu) [inline app/models/course.rb:60]: Is the memoization necessary?

```suggestion
    "https://www.find-postgraduate-teacher-training.service.gov.uk/course/#{provider.code}/#{code}"
```

File: app/services/support_interface/application_monitor.rb [hunk 4]
```diff
@@ -1,21 +1,27 @@
...
-      ApplicationChoice.where(
+      active_applications.where(
         course_option: CourseOption.where(invalidated_by_find: true),
       ).map(&:application_form)
     end
+
+  private
+
+    def active_applications
+      ApplicationChoice.where('status NOT IN (?)', %w[rejected declined withdrawn enrolled])
```
Comment (duncanjbrown) [inline app/services/support_interface/application_monitor.rb:24]:
```suggestion
      ApplicationChoice.where('status NOT IN (?)', %w[rejected declined withdrawn enrolled conditions_not_met])
```

File: app/services/support_interface/application_monitor.rb [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+      ).map(&:application_form)
+    end
+
+    def applications_to_hidden_courses
+      active_applications.where(
+        course_option: CourseOption.joins(:course).where('courses.open_on_apply' => true, 'courses.exposed_in_find' => false),
+      ).map(&:application_form)
+    end
+
+    def applications_to_full_courses
```
Comment (malcolmbaig) [inline app/services/support_interface/application_monitor.rb:15]: I think we have a couple of conflicting concepts here. In hindsight I should've given `invalidated_by_find` a more specific name, but what that field indicates currently is whether a site is still a valid selection for the chosen course, or if it's been removed. That's distinct from whether or not the site is full, which is indicated by `vacancy_status: :no_vacancies`.

"full_courses" is probably best described as any course where all course options are `vacancy_status: :no_vacancies`. At a simpler level, maybe we could make this `applications_to_full_sites` which would query:

```
      active_applications.where(
        course_option: CourseOption.where(vacancy_status: 'no_vacancies'),
      ).map(&:application_form)
```

Comment (tijmenb) [reply app/services/support_interface/application_monitor.rb]: O wow, great catch. I've changed it up here: https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/1834/commits/4746c8823e2f09ef1a2a338f69a0897400b32648

File: spec/services/support_interface/application_monitor_spec.rb [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+      applications = described_class.new.applications_to_full_courses
+
+      expect(applications.map(&:id)).not_to include(with_vacancies.application_form_id)
+      expect(applications.map(&:id)).to include(without_vacancies.application_form_id)
+    end
+  end
+
+  describe '#applications_to_removed_sites' do
+    it 'returns applications to sites that have been removed from Apply' do
```
Comment (malcolmbaig) [inline spec/services/support_interface/application_monitor_spec.rb:48]: nitpick: `removed from Find`, as we don't actively delete sites on our end.

**Comment by malcolmbaig** [comment]
Couple of additional comments, but otherwise LGTM 👍

**Pull request merged**